### DISCUSSION TITLE
Strange error "Battery Monitor - SerialBattery(JKBMS) [1] - Error: Internal calculation error"

### DISCUSSION THREAD

**Discussion by jirkag** [Louisvdw/dbus-serialbattery#1102] [Q&A]
Hi,
I have running installation for long time. I have this alarm already for more than one day. Error appears again after clearing. I have RPI with Venus v3.40, dbus-serialbattery version v1.3.20240705, JKBMS B2A20S20P with 16 cells. 
There is error in console in parameters section.
This error disappears after reboot, but is back in couple of days. This is happening after Venus update and dbus-serialbattery update two weeks ago.
Any idea?
![20240730_143723](https://github.com/user-attachments/assets/ff848f0e-d892-4bf4-9af6-00a65b34c532)

![20240730_143426](https://github.com/user-attachments/assets/1673fde6-8cae-44e3-b363-386a9036fd69)

**Comment by mr-manuel** [comment]
What about checking the logs as the message in `Charge Mode` says?

https://mr-manuel.github.io/venus-os_dbus-serialbattery_docs/troubleshoot/#driver-log-files

**Comment by gnissahr** [comment]
Same here. Cell voltages go missing leading to calculation error. JK BMS connected with usb to TTL. Rebooting or disconnecting/connecting the usb cable seems to solve it, at least for a while. It does not recover by itself.

**Reply by mr-manuel** [reply]
That is correct and expected behavior. If there are errors it will not recover until a restart of the driver or reboot. This for the reason that the user knows there in an error and he has to look into the logs.

**Comment by madtv** [comment]
I have the same issue. I was running 5 batteries together with "dbus-aggregate-batteries" very stable with dbus-serialbattery version v1.2.x.
This "internal calculation error" comes from the JKBMS error "Abnormal Current Sensor" which can be ignored based to the feedback from jkbms. Now after switching to dbus-serialbattery version v1.3.20240705 this jkms error makes the whole system unstable.

**Reply by mr-manuel** [reply]
How often do you get it in the logs in 24h?

**Reply by madtv** [reply]
Very sporadic- 1-2/day and in this case dbus-aggregate stops working because it thinks that there is a heavy issue. Then the victron multiplus system switches the battery off and i cannot charge or discharge the battery anymore until i reboot the whole system/ drivers.
When i switch back to dbus-serialbattery 1.2 everything works, but then my new EEL battery box with jk PB inverter bms will not be recognized anymore, as the bms support starts with 1.3.

**Reply by mr-manuel** [reply]
I will remove that again or change it, that it switches only to error, after 100 errors in 24 hours or something like that.

**Reply by madtv** [reply]
Thank you for the fast response- i try out today!

**Comment by mr-manuel** [comment]
This behaviour was changed with `v1.4.20240806dev`.
